BEIMA.Client/cypress/integration/AddDeviceTypePage.spec.js

```diff
@@ -0,0 +1,90 @@
+/// <reference types="cypress" />
+
+describe("Verify Buttons on Add Device Page", () => {
+  it('Check for Add Device Type Button', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("#addDeviceType").contains('Add Device Type')
+  })
+
+  it('Check for Add Field Button', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("#addField").contains('Add Field')
+  })
+})
+
+describe("Verify Data can be entered into fields", () => {
+  it('Enter data into Name field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get('#Name').type("new type")
+  })
+  it('Enter data into Description field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get('#Description').type("newly added type")
+  })
+  it('Enter data into Device Type Notes field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("[id='Device Type Notes']").type("meter from SEL")
+  })
+})
+
+describe("Verify Data in fields is cleared when Add Device is selected", () => {
+  it('Enter data, click Add Device, verify fields are empty', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get('#Name').type("new type")
+    cy.get('#Description').type("newly added type")
+    cy.get("[id='Device Type Notes']").type("meter from SEL")
+    cy.get("#addDeviceType").scrollIntoView().click()
+    cy.get('#Name').should('have.value', '')
```

Comment:
If we are going to get a field more than once, we should store it in a variable so that it doesn't have to search the DOM again. I think this will help keep the test time down. I might even suggest having some kinds of global variables for the most commonly used ones if that's a possibility.

Reply:
We can't create variable like we are used to using cypress, there's a bit more involved with it. We can grab a DOM component, and have a `then` block following it and call commands on it in there. For places where that was appropriate I updated it, but for something like this where we go back and forth between components we can't use that `then` block.